pkg/autoscaler/scaling/autoscaler.go

```diff
@@ -228,21 +228,29 @@ func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (desiredPodCount
 	// that the deployment does not have enough capacity to serve the desired burst off hand.
 	// EBC = TotCapacity - Cur#ReqInFlight - TargetBurstCapacity
 	excessBCF := -1.
+	// By default we need just one Activator to back the revision.
+	numAct = 1
 	switch {
 	case a.deciderSpec.TargetBurstCapacity == 0:
 		excessBCF = 0
-	case a.deciderSpec.TargetBurstCapacity >= 0:
-		excessBCF = math.Floor(float64(originalReadyPodsCount)*a.deciderSpec.TotalValue - observedStableValue -
+		// numAct stays 1, only needed to scale from 0.
+	case a.deciderSpec.TargetBurstCapacity > 0:
+		totCap := float64(originalReadyPodsCount) * a.deciderSpec.TotalValue
+		excessBCF = math.Floor(totCap - observedStableValue -
 			a.deciderSpec.TargetBurstCapacity)
+		numAct = int32(math.Max(1, math.Ceil((totCap+a.deciderSpec.TargetBurstCapacity)/a.deciderSpec.ActivatorCapacity)))
+	case a.deciderSpec.TargetBurstCapacity == -1:
+		numAct = int32(math.Max(1,
+			math.Ceil(float64(originalReadyPodsCount)*a.deciderSpec.TotalValue/a.deciderSpec.ActivatorCapacity)))
```

Comment:
The value of `numAct` seems like it will take a pretty dramatic hit here.  We'll go from having all activators all the time, to just enough to cover current traffic.

I kinda interpret `TBC=-1` as "everything you've got" mode, so I feel like the "right" value here is to keep all of the activator in for these revisions.  WDYT?

One other thing occurred to me reading through this, which is that these calculations assume that the Revision has exclusive access to the `ActivatorCapacity` for the activators wired in.  Given that we leave activators wired in at low scale (and with TBC=-1 ALL scales), this seems like it could lead to some undesirable results 😬

Reply:
1. _all_ is bad we know that — the LB sucks, especially for small revisions. At this point I mostly assume -1 is asking for  our bespoke load balancing, rather than for burst protection (if you want burst protection set TBC=2000, e.g. — will achieve both).
2. Yes, I've been thinking about this from the beginning. But in general I came to the conclusion that as the # of revisions/pods goes up so does load on the activator and so does its scale. So in general this should cancel each other out.

We can chat more about this :-) But I think this is where I'd like to start and then we can iterate more on the formula.